core/src/main/scala/org/bitcoins/core/crypto/Sign.scala

```diff
@@ -26,27 +27,84 @@ trait Sign {
   def signFuture(bytes: ByteVector): Future[ECDigitalSignature] =
     signFunction(bytes)
 
+  def signWithEntropyFunction: (
+      ByteVector,
+      ByteVector) => Future[ECDigitalSignature]
+
+  def signWithEntropyFuture(
+      bytes: ByteVector,
+      entropy: ByteVector): Future[ECDigitalSignature] =
+    signWithEntropyFunction(bytes, entropy)
+
+  private def signLowRFuture(bytes: ByteVector, startAt: UInt32)(
+      implicit ec: ExecutionContext): Future[ECDigitalSignature] = {
+    val startBytes = startAt.bytes.padLeft(32)
+    val startPlusOneBytes = (startAt + UInt32.one).bytes.padLeft(32)
+
+    val sig1F: Future[ECDigitalSignature] =
+      signWithEntropyFunction(bytes, startBytes)
+    val sig2F: Future[ECDigitalSignature] =
+      signWithEntropyFunction(bytes, startPlusOneBytes)
+    sig1F.flatMap { sig1 =>
+      if (sig1.bytes.length <= 70) {
+        Future.successful(sig1)
+      } else {
+        sig2F.flatMap { sig2 =>
+          if (sig2.bytes.length <= 70) {
```

Comment:
In the bitcoin-core PR they have this
```
// In DER serialization, all values are interpreted as big-endian, signed integers. The highest bit in the integer indicates
// its signed-ness; 0 is positive, 1 is negative. When the value is interpreted as a negative integer, it must be converted
// to a positive value by prepending a 0x00 byte so that the highest bit is 0. We can avoid this prepending by ensuring that
// our highest bit is always 0, and thus we must check that the first byte is less than 0x80.
```

I believe this would be more efficient so we should probably use this

Reply:
Thing is if you look here: https://github.com/bitcoin-s/secp256k1/pull/7/files#diff-0857e942f42747dfba3759e3319abf4fR135

We DER serialize in libsecp256k1 before returning through the JNI so we don't have access to the compact version. So it would be less efficient to do the processing to get this byte